lib/Sabberworm/CSS/Parsing/SourceException.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Sabberworm\CSS\Parsing;
+
+class SourceException extends \Exception {
+    private $iLineNo;
+    public function __construct($sMessage, $iLineNo = 0) {
+        $this->$iLineNo = $iLineNo;
```

Comment:
> On 30.06.2016, at 14:45, Sidharth Kshatriya <notifications@github.com> wrote:
> 
> @sabberworm Why should we remove the $ -- sorry I didn't understand...
> 
> 

It’s just a typo. The line reads `$this->$iLineNo = $iLineNo;` but should be `$this->iLineNo = $iLineNo;`.

If $iLineNo is `1`, the former sets `$this->1` to `1`, the latter sets `$this->iLineNo` to `1`.

The newly-added test `testUnexpectedTokenExceptionLineNo` should have caught this but didn’t because you didn’t `use Sabberworm\CSS\Parsing\UnexpectedTokenException`.